### ISSUE TITLE
debugging obfuscated dll

### ISSUE THREAD

**Issue by mako3551** [yck1509/ConfuserEx#269]
Hi, 

 I am using confuser ex on obfuscating license dll...it works fine... but I need to debug it by myself sometimes... it is possible to debug obfuscated dll? I am attaching on process from Visual studio 2012 and it crashes on Exception: System.ExecutionEngineException

![exception](https://cloud.githubusercontent.com/assets/13132568/8451203/b0c73f60-1fe1-11e5-8ea4-7e234130eb3d.png)

**Comment by yck1509** [comment]
Hi,
To debug the ouput, you need to disable the Anti Debug protection.

**Comment by mako3551** [comment]
yes I tried that (I deleted all the protections)..but stil the exception is throwned...but when I am obfuscating the dll command line writes warning:  Could not find resource type...

![commandline](https://cloud.githubusercontent.com/assets/13132568/8451942/7d9c67fe-1fe7-11e5-84bc-cb614509df86.png)

Maybe the exception is throwned because of this warning

**Comment by yck1509** [comment]
What settings did you used? Also, you might want to try the latest builds from [the CI Server](https://ci.appveyor.com/project/yck1509/confuserex).

**Comment by mako3551** [comment]
I am using version v0.5.0

![protections](https://cloud.githubusercontent.com/assets/13132568/8452127/b11da4e8-1fe8-11e5-950b-43ade278417f.png)

**Comment by yck1509** [comment]
You need to set to preset to a lower value, as the Maximum preset includes all protections, including Anti Debug protection.

**Comment by mako3551** [comment]
Ok I used this settings and I stil have the smae problem...

![protections](https://cloud.githubusercontent.com/assets/13132568/8452366/d6aa6492-1fea-11e5-9cc8-61876cb47314.png)

**Comment by yck1509** [comment]
Anti Debug protection is included in the Minimum preset. In fact, you could keep your previous settings and add a line: "Remove" "anti debug" and it should solve it.

**Comment by mako3551** [comment]
I need to add this line to project xml?

**Comment by yck1509** [comment]
You could do it like this:

![image](https://cloud.githubusercontent.com/assets/846268/8452553/6e31e78a-201e-11e5-8cec-f46a2a9f107a.png)

**Comment by mako3551** [comment]
My settings now: 

![protections](https://cloud.githubusercontent.com/assets/13132568/8452846/a4e50ddc-1fee-11e5-98f5-9fc1fe1278ce.png)


And it crashes on this: 

![exception](https://cloud.githubusercontent.com/assets/13132568/8452857/ae23800e-1fee-11e5-9e95-59cb89d5c84c.png)

**Comment by yck1509** [comment]
Hi,
It doesn't seems to be caused by Anti Debug protection. Does it work outside the debugger?

**Comment by Gaushick** [comment]
We can avoid the "ExecutionEngineException" from the obfuscated dll to remove anti-debug mode.

**Comment by niravparsana94** [comment]
@mako3551  Did you find any solution?

**Comment by mako35551** [comment]
Hi niravparsana94,

no. Situations when I need to debug this dll are very rare, so I decided to let it be. When I need to debug this dll, I need to copy not obfuscated dll to project and start debug.

**Comment by Diemongo** [comment]
MAJ-02-2018:  The following settings will also do the trick:

<project outputDir="Obfuscated" baseDir="." xmlns="http://confuser.codeplex.com">
    <module path="YOUR.dll">
        <rule preset="none" pattern="true">
            <protection id="constants" />
            <protection id="ctrl flow" />
            <protection id="ref proxy" />
        </rule>
    </module>
</project>
